napari/layers/intensity_mixin.py

```diff
@@ -106,8 +110,16 @@ def contrast_limits_range(self):
 
     @contrast_limits_range.setter
     def contrast_limits_range(self, value):
-        """Set the valid range of the contrast limits"""
+        """Set the valid range of the contrast limits.
+        If either value is "None", the current range will be preserved.
+        If the range overlaps the visible contrast limits, the range contrast
+        limits will be clipped to the range of the currently set contrast
```

Comment:
This makes it sound like the new value of `contrast_limits_range` is clipped based on the current value of `current_limits`. But isn't it the other way round?

Reply:
I think I was just not right there. Below is the behavior. I reworded it.

```
layer.contrast_limits
Out[2]: [0, 255]

layer.contrast_limits_range
Out[3]: [0, 255]

layer.contrast_limits = [30, 90]

layer.contrast_limits_range
Out[5]: [0, 255]

layer.contrast_limits_range = [100, 150]

layer.contrast_limits
Out[7]: [100, 150]

layer.contrast_limits_range = [50, 120]

layer.contrast_limits
Out[9]: [100, 120]
```

Reply:
Why doesn't that last command:
`layer.contrast_limits_range = [50, 120]`
make `layer.contrast_limits` be [50, 120] ?

Reply:
Hmm. Yes, that is what we discussed. I must have had a regression at some point. I'll look into it.

Reply:
@psobolewskiPhD I'm not sure how I got those results. I must have been on the wrong branch at the time. The tests are written in the way you have stated that you expect and they are passing (also, manually checking shows the correct expected behavior as well). 

Expected behavior:  
```
        If the requested contrast range limits are completely outside the
        current contrast limits, the range will be set as requested and the
        contrast limits will be reset to the new range.
```

I believe this PR is complete.

Reply:
> @psobolewskiPhD I'm not sure how I got those results. I must have been on the wrong branch at the time. The tests are written in the way you have stated that you expect and they are passing (also, manually checking shows the correct expected behavior as well).
> 
> Expected behavior:
> 
> ```
>         If the requested contrast range limits are completely outside the
>         current contrast limits, the range will be set as requested and the
>         contrast limits will be reset to the new range.
> ```
> 
> I believe this PR is complete.

Sorry, it was actually my bad. I got limits and limits_range confused.
```
layer.contrast_limits
Out[7]: [100, 150]

layer.contrast_limits_range = [50, 120]

layer.contrast_limits
Out[9]: [100, 120]
```
This is the proper behavior. Limits are set, the new range is wider than the bottom limit, so the limit is not reset, just the range. The upper range is more narrow, so that's reset.
And yea, tests pass. Sorry about the confusion!